python/packages/autogen-ext/src/autogen_ext/task/_source_match_termination.py

```diff
@@ -0,0 +1,37 @@
+from typing import Sequence, List
+
+from autogen_agentchat.base import TerminationCondition, TerminatedException
+from autogen_agentchat.messages import StopMessage, AgentMessage
+
+
+class SourceMatchTermination(TerminationCondition):
+    """Terminate the conversation after a specific agent responds.
+
+    Args:
+        agents (List[str]): List of agent names to terminate the conversation.
+
+    Raises:
+        TerminatedException: If the termination condition has already been reached.
+    """
+
+    def __init__(self, agents: List[str]) -> None:
```

Comment:
Since we renamed to source I think this should be sources rather than agents. Agents can refer to the objects or the name, it's a bit ambiguous. It is possible source may not be agent names, such as user.

Reply:
Updated. can you please check again?